### ISSUE TITLE
pgsqld_monitor_0 on <hostname> 'not configured' 

### ISSUE THREAD

**Issue by Netbulae** [dalibo/PAF#49]
Hi,

I followed the guide and documentation, but I keep being unable to run the HA postgresql on Centos 7.2

What can be the cause of:

pgsqld_monitor_0 on database1 'not configured' (6): call=18, status=complete, exitreason='none',
    last-rc-change='Tue Nov  8 14:02:39 2016', queued=0ms, exec=148ms

My ocf:heartbeat:pgsqlms config:
# pgsqld
pcs -f cluster1.xml resource create pgsqld ocf:heartbeat:pgsqlms \
    bindir=/usr/pgsql-9.6/bin pgdata=/var/lib/pgsql/9.6/data     \
    op start timeout=60s                                         \
    op stop timeout=60s                                          \
    op promote timeout=30s                                       \
    op demote timeout=120s                                       \
    op monitor interval=15s timeout=10s role="Master"            \
    op monitor interval=16s timeout=10s role="Slave"             \
    op notify timeout=60s                                        \

My "pcs status" output:

Cluster name: cluster_pgsql
Last updated: Tue Nov  8 14:02:44 2016		Last change: Tue Nov  8 14:02:37 2016 by root via cibadmin on database1
Stack: corosync
Current DC: database1 (version 1.1.13-10.el7_2.4-44eb2dd) - partition with quorum
3 nodes and 7 resources configured

Online: [ database1 database2 database3 ]

Full list of resources:

 fence_vm_database1	(stonith:fence_rhevm):	Started database2
 fence_vm_database2	(stonith:fence_rhevm):	Stopped
 fence_vm_database3	(stonith:fence_rhevm):	Stopped
 Master/Slave Set: psql-ha [pgsqld]
     pgsqld	(ocf::heartbeat:pgsqlms):	FAILED database1
     pgsqld	(ocf::heartbeat:pgsqlms):	FAILED database2
     pgsqld	(ocf::heartbeat:pgsqlms):	FAILED database3
 pgsql-master-ip	(ocf::heartbeat:IPaddr2):	Stopped

Failed Actions:
* pgsqld_monitor_0 on database1 'not configured' (6): call=18, status=complete, exitreason='none',
    last-rc-change='Tue Nov  8 14:02:39 2016', queued=0ms, exec=148ms
* pgsqld_monitor_0 on database2 'not configured' (6): call=18, status=complete, exitreason='none',
    last-rc-change='Tue Nov  8 14:02:40 2016', queued=0ms, exec=297ms
* pgsqld_monitor_0 on database3 'not configured' (6): call=18, status=complete, exitreason='none',
    last-rc-change='Tue Nov  8 14:02:40 2016', queued=0ms, exec=304ms

PCSD Status:
  database1: Online
  database2: Online
  database3: Online

Daemon Status:
  corosync: active/disabled
  pacemaker: active/disabled
  pcsd: active/enabled

**Comment by ioguix** [comment]
Hello @Netbulae,

There's various ways PAF could refuse to deal with your setup. Usually it explains what is wrong in log files so you can fix the issue on your side. See here for the various checks PAF is doing in 2.0: https://github.com/dalibo/PAF/blob/v2.0.0/script/pgsqlms#L984

However, maybe you felt in the only known incompatibility between PAF and PostgreSQL 9.6: how did you set your `wal_level` parameter? We accepted a pull request few days ago about `wal_level = replica`, see #47.

This fix will be included in the next release.

Regards,

**Comment by ioguix** [comment]
Any news on this issue? Should I close it?

**Comment by Netbulae** [comment]
Hi, still the same issue. Trying to find anything in the log. What I can see is:

Nov 21 15:57:29 [3494] database1.test.nu   pengine:  warning: unpack_rsc_op_failure:	Processing failed op stop for pgsqld:0 on database1: not configured (6)
Nov 21 15:57:29 [3494] database1.test.nu   pengine:    error: unpack_rsc_op:	Preventing psql-ha from re-starting anywhere: operation stop failed 'not configured' (6)
Nov 21 15:57:29 [3494] database1.test.nu    pengine:  warning: unpack_rsc_op_failure:	Processing failed op stop for pgsqld:0 on database1: not configured (6)
Nov 21 15:57:29 [3494] database1.test.nu    pengine:    error: unpack_rsc_op:	Preventing psql-ha from re-starting anywhere: operation stop failed 'not configured' (6)

For now, I have both slave servers set to hot standby:
wal_level = hot_standby

**Comment by ioguix** [comment]
Hi,

Even if you set `wal_level = hot_standby` in postrgesql.conf, this parameter is set to `replica` internally.

The following command expose this incompatibility with current PAF 2.0:
```
$ psql -c "show wal_level"
 wal_level 
-----------
 replica
```

I'm afraid you either need to use current HEAD code, patch pgsqlms yourself using #47 or wait for release 2.1.

**Comment by ioguix** [comment]
I'm closing this issue as this has been fixed in #47 and will be included in next release.

Thank you for reporting. Do not hesitate to ask more questions if needed.
